### PR TITLE
feat: add Dockerfile

### LINKED ISSUE #245: support official docker image
First of all, thanks for the great toolkit!
I'll be happy to contribute code, if this is something you consider doing.

**Is your feature request related to a problem? Please describe.**

Currently, there is no officially supported docker image, and I don't particularly like to install npm only to use this amazing cli tool.

**Describe the solution you'd like**

I would prefer if the openapi-cli tool would additionally be distributed as a docker image.

**Describe alternatives you've considered**

Alternatively, openapi-cli could be packaged as standalone executable for each target platform.
But that means more maintenance overhead for releasing.

### PR DESCRIPTION
**Pull request by Oncilla**
Add a Dockerfile to allow running the openapi-cli tool in a docker container.

fixes #245

---
This change is [Reviewable](https://reviewable.io/reviews/redocly/openapi-cli/247)

### CODE REVIEW

**Comment by Marshevskyy** [comment]
Looks good. Could you update readme file as well?

**Comment by Oncilla** [comment]
Added a a small description to the readme file.

I also added a GitHub action that will automatically deploy the docker image to Docker Hub.
That should reduce your maintenance burden.

Note that you will need to [add the following two secrets](https://docs.github.com/en/actions/reference/encrypted-secrets): 

```
secrets.DOCKERHUB_USERNAME
secrets.DOCKERHUB_TOKEN
```

File: README.md [hunk 1]
```diff
@@ -29,12 +29,32 @@ Currently, @redocly/openapi-cli supports these features:
...
+```
 
+To build and run with a local image, run the following from the project root:
+
+```
+docker build -t openapi-cli .
+docker run --rm -v $PWD:/spec oepnapi-cli lint path-to-root-file.yaml
+```
+
+## [Read the docs](https://redoc.ly/docs/cli/)
```
Comment (waldyrious) [inline README.md:57]: Was the removal of the blank line below this one an intentional change?

Comment (Oncilla) [reply README.md:57]: Done.

File: .github/workflows/dockerhub.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# https://github.com/marketplace/actions/build-and-push-docker-images#handle-tags-and-labels
+
+name: CD to Docker Hub
+
+on:
+  push:
+    tags:
+      - v[0-9]*.[0-9]*.[0-9]*
```
Comment (RomanHotsiy) [inline .github/workflows/dockerhub.yaml:8]:
```suggestion
      - v[0-9]+.[0-9]+.[0-9]+
      - v[0-9]+.[0-9]+.[0-9]+-[a-z]+.[0-9]+ # beta/rc releases
```

File: .github/workflows/dockerhub.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          images: redocly/openapi-cli
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        if: github.event_name != 'pull_request'
```
Comment (RomanHotsiy) [inline .github/workflows/dockerhub.yaml:30]: Looks like this is extra as this workflow runs only on tags push->tags anyway.

```suggestion
```

File: .github/workflows/dockerhub.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: ./Dockerfile
+          platforms: linux/amd64,linux/arm64,linux/386
+          push: ${{ github.event_name != 'pull_request' }}
```
Comment (RomanHotsiy) [inline .github/workflows/dockerhub.yaml:42]:
```suggestion
          push: true
```

File: README.md [hunk 5]
```diff
@@ -29,9 +29,30 @@ Currently, @redocly/openapi-cli supports these features:
...
+
+```
+docker run --rm -v $PWD:/spec redocly/openapi-cli lint path-to-root-file.yaml
+```
+
+To build and run with a local image, run the following from the project root:
+
+```
+docker build -t openapi-cli .
+docker run --rm -v $PWD:/spec oepnapi-cli lint path-to-root-file.yaml
```
Comment (RomanHotsiy) [inline README.md:54]:
```suggestion
docker run --rm -v $PWD:/spec openapi-cli lint path-to-root-file.yaml
```

**Review by RomanHotsiy — APPROVED** [review]

**Comment by RomanHotsiy** [comment]
Thanks, @Oncilla!

Sorry for the huge delay. I just pushed the latest version manually: https://hub.docker.com/repository/docker/redocly/openapi-cli

The next version should be pushed by the GitHub action when we cut a new release.

**Comment by Oncilla** [comment]
No worries, I know the struggle 👍

**Pull request merged**